main.rb

```diff
@@ -0,0 +1,160 @@
+    my_each do |x|
+      if block_given?
+        return false unless yield x
+      elsif pattern.class == Regexp
+        return false unless pattern =~ x
+      elsif pattern.class == Class
+        return false unless x.class == pattern
+      elsif !pattern.nil?
+        return false unless x == pattern
+      else
+        return false unless x
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    return false if (self - [nil, false]) == []
+
+    my_each do |x|
+      if block_given?
+        return true if yield x
+      elsif pattern.class == Regexp
+        return true if pattern =~ x
+      elsif pattern.class == Class
+        return true if x.class == pattern
+      elsif !pattern.nil?
+        return true if x == pattern
+      else
+        return true unless x
+      end
+    end
+    return true if pattern.nil? && !block_given?
+
+    false
+  end
+
+  def my_none?(pattern = nil)
+    return true if (self - [nil, false]) == [] || length < 2
+
+    my_each_with_index do |x, i|
+      if block_given?
+        return false if yield x
+      elsif pattern.class == Regexp
+        return false if pattern =~ x
+      elsif pattern.class == Class
+        return false if x.class == pattern
+      elsif !pattern.nil?
+        return false if x == pattern
+      elsif i.positive? && self[i] != self[i - 1]
+        return false
+      end
+    end
+    true
+  end
+
+  # rubocop: enable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+  def my_count(arg = nil)
+    return length if arg.nil? && !block_given?
+
+    count = 0
+    my_each do |x|
+      if block_given?
+        count += 1 if yield x
+      elsif x == arg
+        count += 1
+      end
+    end
+
+    count
+  end
+
+  def my_map(proc = nil)
+    return to_enum unless block_given?
+
+    arr = []
+    to_a.my_each { |x| arr << (!proc.nil? ? proc.call(x) : yield(x)) }
+    arr
+  end
+
+  # rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+  def my_inject(start = nil, arg = nil)
+    arr = self
+    if arg == nil? && block_given?
+      result = start
+      arr.my_each do |i|
+        result =
+          if result.nil?
+            i
+          else
+            yield(result, i)
+          end
+      end
+    elsif (start.class != Symbol && arg.nil?) && start.class ==
+                                                 Integer
+      warn "The value #{start} is not a symbol rep"
+      abort
+    elsif start.class == Symbol
+      if start == :+
+        result = arr.my_inject { |i, v| i + v }
+      elsif start == :*
+        result = arr.my_inject { |i, v| i * v }
+      elsif start == :-
+        result = arr.my_inject { |i, v| i - v }
+      elsif start == :/
+        result = arr.my_inject { |i, v| i / v }
+      end
+    elsif start.class == Integer && arg.class == Symbol
+      new_arr = arr.to_a
+      new_arr.unshift(start)
+      result = new_arr.my_inject(arg)
+    end
+    result
+  end

```

Comment:
- [ ] my_inject raises a "LocalJumpError" when no block or argument is given

Example:

`[1, 2, 3, 4, 5].inject` should raise a LocalJumpError

Print original method:
![Screenshot from 2020-06-05 10-02-24](https://user-images.githubusercontent.com/30977845/83879242-b5879400-a713-11ea-9443-4fac8b0ac675.png)

- [ ] my_inject, when a block is given without an initial value, combines all elements of enum by applying a binary operation, specified by a block

Example: 

`[1, 2, 3, 4, 5].my_inject { |sum, value| sum + value }` should return 15
`[1, 2, 3, 4, 5].my_inject { |product, value| product * value }` should return 120
`[1, 2, 3, 4, 5].my_inject { |sub, value| sub - value }` should return 13

Should also work for Range:

`(1..5).inject { |sum, value| sum + value }` should return 15